internal/datastore/postgres/migrations/zz_migration.0020_add_watch_api_index.go

```diff
@@ -0,0 +1,23 @@
+package migrations
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/jackc/pgx/v5"
+)
+
+const addWatchAPIIndexToRelationTupleTable = `CREATE INDEX CONCURRENTLY IF NOT EXISTS ix_watch_index ON relation_tuple (created_xid);`
```

Comment:
just a thought for later: since the query also filters on `deleted_xid`, we might want to try a composite index on (created_xid, deleted_xid) down the line.

`CREATE INDEX CONCURRENTLY IF NOT EXISTS ix_watch_index ON relation_tuple (created_xid, deleted_xid);`

It could make queries with both columns even faster by improving how rows are filtered. Definitely not a blocker tho.